### ISSUE TITLE
Rustdoc cann't build rust documentation with custom libdir

### ISSUE THREAD

**Issue by o01eg** [rust-lang/rust#58587]
After https://github.com/rust-lang/rust/pull/58238 it was broken again:

```
Documenting stage2 std (x86_64-unknown-linux-gnu)
      c Assemble { target_compiler: Compiler { stage: 2, host: "x86_64-unknown-linux-gnu" } }
      c Assemble { target_compiler: Compiler { stage: 1, host: "x86_64-unknown-linux-gnu" } }
      c Std { target: "x86_64-unknown-linux-gnu", compiler: Compiler { stage: 1, host: "x86_64-unknown-linux-gnu" } }
      c Rustdoc { host: "x86_64-unknown-linux-gnu" }
      c Sysroot { compiler: Compiler { stage: 1, host: "x86_64-unknown-linux-gnu" } }
      c Sysroot { compiler: Compiler { stage: 1, host: "x86_64-unknown-linux-gnu" } }
      c Sysroot { compiler: Compiler { stage: 1, host: "x86_64-unknown-linux-gnu" } }
      c Rustdoc { host: "x86_64-unknown-linux-gnu" }
      c Sysroot { compiler: Compiler { stage: 1, host: "x86_64-unknown-linux-gnu" } }
      c Llvm { target: "x86_64-unknown-linux-gnu", emscripten: false }
running: "/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage0/bin/cargo" "rustdoc" "--target" "x86_64-unknown-linux-gnu" "-j" "7" "-v" "--release" "--locked" "--features" "panic-unwind backtrace" "--manifest-path" "/tmp/portage/dev-lang/rust-9999/work/rust-git-src/src/libstd/Cargo.toml" "-Z" "unstable-options" "-p" "alloc" "--" "--markdown-css" "rust.css" "--markdown-no-toc" "--generate-redirect-pages" "--index-page" "/tmp/portage/dev-lang/rust-9999/work/rust-git-src/src/doc/index.md"
       Fresh cc v1.0.28
    Checking core v0.0.0 (/tmp/portage/dev-lang/rust-9999/work/rust-git-src/src/libcore)
     Running `/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/bootstrap/debug/rustc --crate-name core src/libcore/lib.rs --color always --crate-type lib --emit=dep-info,metadata -C opt-level=2 -C metadata=3976fdee207fc816 -C extra-filename=-3976fdee207fc816 --out-dir /tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1-std/x86_64-unknown-linux-gnu/release/deps --target x86_64-unknown-linux-gnu -L dependency=/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1-std/x86_64-unknown-linux-gnu/release/deps -L dependency=/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1-std/release/deps`
rustc command: "LD_LIBRARY_PATH"="/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1/lib:/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1-std/release/deps:/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage0/lib" "/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1/bin/rustc" "--crate-name" "core" "src/libcore/lib.rs" "--color" "always" "--crate-type" "lib" "--emit=dep-info,metadata" "-C" "opt-level=2" "-C" "metadata=3976fdee207fc816-rustc" "-C" "extra-filename=-3976fdee207fc816" "--out-dir" "/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1-std/x86_64-unknown-linux-gnu/release/deps" "--target" "x86_64-unknown-linux-gnu" "-L" "dependency=/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1-std/x86_64-unknown-linux-gnu/release/deps" "-L" "dependency=/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1-std/release/deps" "--cfg" "stage1" "--sysroot" "/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1" "-Cprefer-dynamic" "-Clinker=x86_64-pc-linux-gnu-gcc" "-C" "debug-assertions=n" "-Z" "force-unstable-if-unmarked" "-Dwarnings" "-Dbare_trait_objects"
sysroot: "/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1"
libdir: "/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1/lib"
    Checking rustc-std-workspace-core v1.0.0 (/tmp/portage/dev-lang/rust-9999/work/rust-git-src/src/tools/rustc-std-workspace-core)
     Running `/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/bootstrap/debug/rustc --edition=2018 --crate-name rustc_std_workspace_core src/tools/rustc-std-workspace-core/lib.rs --color always --crate-type lib --emit=dep-info,metadata -C opt-level=2 -C metadata=5b71f57eb44caeb5 -C extra-filename=-5b71f57eb44caeb5 --out-dir /tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1-std/x86_64-unknown-linux-gnu/release/deps --target x86_64-unknown-linux-gnu -L dependency=/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1-std/x86_64-unknown-linux-gnu/release/deps -L dependency=/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1-std/release/deps --extern core=/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1-std/x86_64-unknown-linux-gnu/release/deps/libcore-3976fdee207fc816.rmeta`
rustc command: "LD_LIBRARY_PATH"="/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1/lib:/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1-std/release/deps:/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage0/lib" "/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1/bin/rustc" "--edition=2018" "--crate-name" "rustc_std_workspace_core" "src/tools/rustc-std-workspace-core/lib.rs" "--color" "always" "--crate-type" "lib" "--emit=dep-info,metadata" "-C" "opt-level=2" "-C" "metadata=5b71f57eb44caeb5-rustc" "-C" "extra-filename=-5b71f57eb44caeb5" "--out-dir" "/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1-std/x86_64-unknown-linux-gnu/release/deps" "--target" "x86_64-unknown-linux-gnu" "-L" "dependency=/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1-std/x86_64-unknown-linux-gnu/release/deps" "-L" "dependency=/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1-std/release/deps" "--extern" "core=/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1-std/x86_64-unknown-linux-gnu/release/deps/libcore-3976fdee207fc816.rmeta" "--cfg" "stage1" "--sysroot" "/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1" "-Cprefer-dynamic" "-Clinker=x86_64-pc-linux-gnu-gcc" "-C" "debug-assertions=n" "-Z" "force-unstable-if-unmarked" "-Dwarnings" "-Dbare_trait_objects"
sysroot: "/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1"
libdir: "/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1/lib"
    Checking compiler_builtins v0.1.5
     Running `/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/bootstrap/debug/rustc --crate-name compiler_builtins /tmp/portage/dev-lang/rust-9999/homedir/.cargo/registry/src/github.com-1ecc6299db9ec823/compiler_builtins-0.1.5/src/lib.rs --color always --crate-type lib --emit=dep-info,metadata -C opt-level=2 --cfg 'feature="c"' --cfg 'feature="cc"' --cfg 'feature="compiler-builtins"' --cfg 'feature="core"' --cfg 'feature="default"' --cfg 'feature="rustc-dep-of-std"' -C metadata=52d220f7a884ebdb -C extra-filename=-52d220f7a884ebdb --out-dir /tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1-std/x86_64-unknown-linux-gnu/release/deps --target x86_64-unknown-linux-gnu -L dependency=/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1-std/x86_64-unknown-linux-gnu/release/deps -L dependency=/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1-std/release/deps --extern core=/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1-std/x86_64-unknown-linux-gnu/release/deps/librustc_std_workspace_core-5b71f57eb44caeb5.rmeta --cap-lints allow -L native=/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1-std/x86_64-unknown-linux-gnu/release/build/compiler_builtins-db1ec1e745c28d7d/out --cfg use_c -l static=compiler-rt`
rustc command: "LD_LIBRARY_PATH"="/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1/lib:/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1-std/release/deps:/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage0/lib" "/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1/bin/rustc" "--crate-name" "compiler_builtins" "/tmp/portage/dev-lang/rust-9999/homedir/.cargo/registry/src/github.com-1ecc6299db9ec823/compiler_builtins-0.1.5/src/lib.rs" "--color" "always" "--crate-type" "lib" "--emit=dep-info,metadata" "-C" "opt-level=2" "--cfg" "feature=\"c\"" "--cfg" "feature=\"cc\"" "--cfg" "feature=\"compiler-builtins\"" "--cfg" "feature=\"core\"" "--cfg" "feature=\"default\"" "--cfg" "feature=\"rustc-dep-of-std\"" "-C" "metadata=52d220f7a884ebdb-rustc" "-C" "extra-filename=-52d220f7a884ebdb" "--out-dir" "/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1-std/x86_64-unknown-linux-gnu/release/deps" "--target" "x86_64-unknown-linux-gnu" "-L" "dependency=/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1-std/x86_64-unknown-linux-gnu/release/deps" "-L" "dependency=/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1-std/release/deps" "--extern" "core=/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1-std/x86_64-unknown-linux-gnu/release/deps/librustc_std_workspace_core-5b71f57eb44caeb5.rmeta" "--cap-lints" "allow" "-L" "native=/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1-std/x86_64-unknown-linux-gnu/release/build/compiler_builtins-db1ec1e745c28d7d/out" "--cfg" "use_c" "-l" "static=compiler-rt" "--cfg" "stage1" "--sysroot" "/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1" "-Cprefer-dynamic" "-Clinker=x86_64-pc-linux-gnu-gcc" "-C" "panic=abort" "-C" "debug-assertions=no" "-Z" "force-unstable-if-unmarked" "-Dwarnings" "-Dbare_trait_objects"
sysroot: "/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1"
libdir: "/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1/lib"
 Documenting alloc v0.0.0 (/tmp/portage/dev-lang/rust-9999/work/rust-git-src/src/liballoc)
     Running `/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/bootstrap/debug/rustdoc --edition=2018 --crate-name alloc src/liballoc/lib.rs --target x86_64-unknown-linux-gnu -o /tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1-std/x86_64-unknown-linux-gnu/doc --markdown-css rust.css --markdown-no-toc --generate-redirect-pages --index-page /tmp/portage/dev-lang/rust-9999/work/rust-git-src/src/doc/index.md -L dependency=/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1-std/x86_64-unknown-linux-gnu/release/deps -L dependency=/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1-std/release/deps --extern compiler_builtins=/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1-std/x86_64-unknown-linux-gnu/release/deps/libcompiler_builtins-52d220f7a884ebdb.rmeta --extern core=/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1-std/x86_64-unknown-linux-gnu/release/deps/libcore-3976fdee207fc816.rmeta`
rustdoc command: "/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage2/bin/rustdoc" "--edition=2018" "--crate-name" "alloc" "src/liballoc/lib.rs" "--target" "x86_64-unknown-linux-gnu" "-o" "/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1-std/x86_64-unknown-linux-gnu/doc" "--markdown-css" "rust.css" "--markdown-no-toc" "--generate-redirect-pages" "--index-page" "/tmp/portage/dev-lang/rust-9999/work/rust-git-src/src/doc/index.md" "-L" "dependency=/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1-std/x86_64-unknown-linux-gnu/release/deps" "-L" "dependency=/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1-std/release/deps" "--extern" "compiler_builtins=/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1-std/x86_64-unknown-linux-gnu/release/deps/libcompiler_builtins-52d220f7a884ebdb.rmeta" "--extern" "core=/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1-std/x86_64-unknown-linux-gnu/release/deps/libcore-3976fdee207fc816.rmeta" "--cfg" "stage1" "--cfg" "dox" "--sysroot" "/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1" "-Z" "force-unstable-if-unmarked" "--linker" "x86_64-pc-linux-gnu-gcc" "-Z" "unstable-options" "-Z" "unstable-options" "--crate-version" "1.34.0-dev (74e35d270 2019-02-19)"
libdir: "/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1/lib"
/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage2/bin/rustdoc: error while loading shared libraries: librustc_driver-ed4f3bf6d6835e3f.so: cannot open shared object file: No such file or directory
error: Could not document `alloc`.

Caused by:
  process didn't exit successfully: `/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/bootstrap/debug/rustdoc --edition=2018 --crate-name alloc src/liballoc/lib.rs --target x86_64-unknown-linux-gnu -o /tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1-std/x86_64-unknown-linux-gnu/doc --markdown-css rust.css --markdown-no-toc --generate-redirect-pages --index-page /tmp/portage/dev-lang/rust-9999/work/rust-git-src/src/doc/index.md -L dependency=/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1-std/x86_64-unknown-linux-gnu/release/deps -L dependency=/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1-std/release/deps --extern compiler_builtins=/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1-std/x86_64-unknown-linux-gnu/release/deps/libcompiler_builtins-52d220f7a884ebdb.rmeta --extern core=/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1-std/x86_64-unknown-linux-gnu/release/deps/libcore-3976fdee207fc816.rmeta` (exit code: 127)


command did not execute successfully: "/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage0/bin/cargo" "rustdoc" "--target" "x86_64-unknown-linux-gnu" "-j" "7" "-v" "--release" "--locked" "--features" "panic-unwind backtrace" "--manifest-path" "/tmp/portage/dev-lang/rust-9999/work/rust-git-src/src/libstd/Cargo.toml" "-Z" "unstable-options" "-p" "alloc" "--" "--markdown-css" "rust.css" "--markdown-no-toc" "--generate-redirect-pages" "--index-page" "/tmp/portage/dev-lang/rust-9999/work/rust-git-src/src/doc/index.md"
expected success, got: exit code: 101


Traceback (most recent call last):
  File "./x.py", line 11, in <module>
    bootstrap.main()
  File "/tmp/portage/dev-lang/rust-9999/work/rust-git-src/src/bootstrap/bootstrap.py", line 838, in main
    bootstrap(help_triggered)
  File "/tmp/portage/dev-lang/rust-9999/work/rust-git-src/src/bootstrap/bootstrap.py", line 824, in bootstrap
    run(args, env=env, verbose=build.verbose)
  File "/tmp/portage/dev-lang/rust-9999/work/rust-git-src/src/bootstrap/bootstrap.py", line 141, in run
    raise RuntimeError(err)
RuntimeError: failed to run: /tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/bootstrap/debug/bootstrap install --verbose --config=/tmp/portage/dev-lang/rust-9999/work/rust-git-src/config.toml -j7
```

**Comment by o01eg** [comment]
Real `librustc_driver-ed4f3bf6d6835e3f.so` placed in
```
/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage2/lib/librustc_driver-ed4f3bf6d6835e3f.so
/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1/lib64/rust-9999/rustlib/x86_64-unknown-linux-gnu/lib/librustc_driver-ed4f3bf6d6835e3f.so
/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1-rustc/x86_64-unknown-linux-gnu/release/deps/librustc_driver-ed4f3bf6d6835e3f.so
```

**Comment by o01eg** [comment]
I've tried to fix it like #52439 but it still failed:
```
 Documenting alloc v0.0.0 (/tmp/portage/dev-lang/rust-9999/work/rust-git-src/src/liballoc)
     Running `/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/bootstrap/debug/rustdoc --edition=2018 --crate-name alloc src/liballoc/lib.rs --target wasm32-unknown-unknown -o /tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage2-std/wasm32-unknown-unknown/doc --markdown-css rust.css --markdown-no-toc --generate-redirect-pages --index-page /tmp/portage/dev-lang/rust-9999/work/rust-git-src/src/doc/index.md -L dependency=/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage2-std/wasm32-unknown-unknown/release/deps -L dependency=/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage2-std/release/deps --extern compiler_builtins=/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage2-std/wasm32-unknown-unknown/release/deps/libcompiler_builtins-52d220f7a884ebdb.rmeta --extern core=/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage2-std/wasm32-unknown-unknown/release/deps/libcore-3976fdee207fc816.rmeta`
rustdoc command: "/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage2/bin/rustdoc" "--edition=2018" "--crate-name" "alloc" "src/liballoc/lib.rs" "--target" "wasm32-unknown-unknown" "-o" "/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage2-std/wasm32-unknown-unknown/doc" "--markdown-css" "rust.css" "--markdown-no-toc" "--generate-redirect-pages" "--index-page" "/tmp/portage/dev-lang/rust-9999/work/rust-git-src/src/doc/index.md" "-L" "dependency=/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage2-std/wasm32-unknown-unknown/release/deps" "-L" "dependency=/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage2-std/release/deps" "--extern" "compiler_builtins=/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage2-std/wasm32-unknown-unknown/release/deps/libcompiler_builtins-52d220f7a884ebdb.rmeta" "--extern" "core=/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage2-std/wasm32-unknown-unknown/release/deps/libcore-3976fdee207fc816.rmeta" "--cfg" "stage2" "--cfg" "dox" "--sysroot" "/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage2" "-Z" "force-unstable-if-unmarked" "--linker" "rust-lld" "-Z" "unstable-options" "-Z" "unstable-options" "--crate-version" "1.34.0-dev (74e35d270 2019-02-19)"
libdir: "/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage2/lib64/rust-9999/rustlib/x86_64-unknown-linux-gnu/lib"
/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage2/bin/rustdoc: error while loading shared libraries: librustc_driver-ed4f3bf6d6835e3f.so: cannot open shared object file: No such file or directory
error: Could not document `alloc`.
```

Looks like one installation of library is missed: https://github.com/rust-lang/rust/issues/52317#issuecomment-405258339

**Comment by o01eg** [comment]
Cann't found which step produces `build/x86_64-unknown-linux-gnu/stage2/{LIBDIR}/rustlib/x86_64-unknown-linux-gnu/lib/librustc_driver-*.so`

**Comment by o01eg** [comment]
Looks like something deletes required librustc_driver-*.so, because after `compile` command I get:
```
# find /tmp/portage/ -name librustc_driver-*.so
/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage2/lib64/rust-9999/rustlib/x86_64-unknown-linux-gnu/lib/librustc_driver-ed4f3bf6d6835e3f.so
/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage2/lib/librustc_driver-ed4f3bf6d6835e3f.so
/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1/lib64/rust-9999/rustlib/x86_64-unknown-linux-gnu/lib/librustc_driver-ed4f3bf6d6835e3f.so
/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1/lib/librustc_driver-a90db68b145e208b.so
/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage0-sysroot/lib/rustlib/x86_64-unknown-linux-gnu/lib/librustc_driver-a90db68b145e208b.so
/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1-rustc/x86_64-unknown-linux-gnu/release/deps/librustc_driver-ed4f3bf6d6835e3f.so
/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage0-rustc/x86_64-unknown-linux-gnu/release/deps/librustc_driver-a90db68b145e208b.so
/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage0/lib/librustc_driver-5c2972ea809e01ae.so
/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage0/lib/rustlib/x86_64-unknown-linux-gnu/lib/librustc_driver-5c2972ea809e01ae.so
```

But in `install` command when I breaks on documenting `alloc` 

```
# find /tmp/portage/ -name librustc_driver-*.so
/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage2/lib/librustc_driver-ed4f3bf6d6835e3f.so
/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1/lib64/rust-9999/rustlib/x86_64-unknown-linux-gnu/lib/librustc_driver-ed4f3bf6d6835e3f.so
/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1/lib/librustc_driver-a90db68b145e208b.so
/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage0-sysroot/lib/rustlib/x86_64-unknown-linux-gnu/lib/librustc_driver-a90db68b145e208b.so
/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage1-rustc/x86_64-unknown-linux-gnu/release/deps/librustc_driver-ed4f3bf6d6835e3f.so
/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage0-rustc/x86_64-unknown-linux-gnu/release/deps/librustc_driver-a90db68b145e208b.so
/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage0/lib/librustc_driver-5c2972ea809e01ae.so
/tmp/portage/dev-lang/rust-9999/work/rust-git-src/build/x86_64-unknown-linux-gnu/stage0/lib/rustlib/x86_64-unknown-linux-gnu/lib/librustc_driver-5c2972ea809e01ae.so
```

**Comment by o01eg** [comment]
Looks like it mostly breaks builds with multiple `build.target`.

**Comment by o01eg** [comment]
Fixed by #58897.
